libcxx/docs/CodingGuidelines.rst

```diff
@@ -184,3 +184,13 @@ headers (which is sometimes required for ``constexpr`` support).
 
 When defining a function at the ABI boundary, it can also be useful to consider which attributes (like ``[[gnu::pure]]``
 and ``[[clang::noescape]]``) can be added to the function to improve the compiler's ability to optimize.
+
+library-internal type aliases should be annotated with ``_LIBCPP_NODEBUG``
+==========================================================================
+
+Libc++ has lots of internal type aliases. Accumulated, these can result in significant amounts of debug information that
+users don't care about usually, since users don't try to debug standard library facilities in most cases. For that
+reason, all library-internal type aliases should be annotated with ``_LIBCPP_NODEBUG`` to suppress compilers from
```

Comment:
It seems that function-local type aliases should be exempted, right? Also, it seems intended that member type alias `type` should also be marked `_LIBCPP_NODEBUG` even when it is not internal.

Reply:
`type` is a follow-up. I'll update the documentation when we actually do it consistently.